notebooks/article_1.py

```diff
@@ -0,0 +1,1093 @@
+            "y": data_df["age_last_funding_year"],
+            "label": "Difference 'Founded' and 'Last Funding'"
+        },
+
+    ]
+
+    # Create each scatter plot
+    for i, config in enumerate(plot_configs, start=1):
+        plt.subplot(1, 4, i)
+        sns.scatterplot(x=config["x"], y=config["y"], hue=data_df["is_outlier"], palette=custom_palette)
+        if config["label"]:
+            plt.xlabel(config["label"])
+
+    plt.tight_layout()  #Adjust the layout
+    plt.show()
+
+plot_outlier_scatterplots(dataframe)
+
+
+# All negative values are successfully removed!
+#
+# This time, we focus on the positive outliers (the red points) that were identified using the **z-score** method in our visualization. Let's proceed to analyze this method in detail.
+
+#  ### Outliers
+
+# Outliers are data points that stand out from the rest of the data because they are much higher or lower than most of the other values in the dataset. To detect these outliers, we'll use two effective tools:
+#
+# - **Histograms** These are visual representations that help us understand the distribution of our data. By plotting the frequency of data points, histograms make it easier to spot any values that fall far outside the typical range.
+#
+# Let's determine which variables have the highest number of outliers.
+
+# In[607]:
+
+
+# Select only numeric columns for histogram plots
+specific_variables = [
+    "age_first_funding_year", "age_last_funding_year",
+    "age_first_milestone_year", "age_last_milestone_year",
+    "funding_total_usd", "funding_rounds",
+    "milestones", "relationships"
+]
+# Determine the number of rows and columns for the subplots
+num_variables = len(specific_variables)
+num_columns = 5
+num_rows = int(np.ceil(num_variables / num_columns))
+
+# Set the overall figure size
+fig_width = num_columns * 6
+fig_height = num_rows * 5
+
+plt.figure(figsize=(fig_width, fig_height))
+
+for i, column in enumerate(specific_variables):
+    plt.subplot(num_rows, num_columns, i+1)
+    plt.title(column, fontsize=25)
+    sns.histplot(dataframe[column], color="red", kde=True)  # Adding KDE for smooth distribution curve.
+    plt.xlabel('')
+    plt.ylabel('')
+
+# Adjust layout to prevent overlap
+plt.tight_layout()
+plt.show()
+
+
+# As observed in the histogram, noticeable spikes are apparent in the data. Excluding binary variables, the variables with the most prominent spikes are:
+#
+# - `age_first_funding_year`
+# - `age_first_milestone_year`
+# - `age_last_milestone_year`
+# - `funding_total_usd`
+# - `relationships`
+
+# Now that we have identified outliers in our data, the next step is normalization. We will utilize the log-transformed method for this purpose.
+
+# #### Handle Outliers
+
+# To handle this outliers we will use **The log-transformed method** that involves taking the natural logarithm (ln) of each data point in a dataset. It's useful when data has a wide range of values, helping to balance the data.
```

Comment:
But by computing the log of the data, we change its nature. Can you check if this will be detrimental for the analysis please?